lib/active_model_serializers/adapter/json_api.rb

```diff
@@ -43,14 +44,13 @@ def default_key_transform
 
       # {http://jsonapi.org/format/#crud Requests are transactional, i.e. success or failure}
       # {http://jsonapi.org/format/#document-top-level data and errors MUST NOT coexist in the same document.}
-      def serializable_hash(options = nil)
-        options ||= {}
+      def serializable_hash(_options = nil)
         document = if serializer.success?
-                     success_document(options)
+                     success_document
                    else
                      failure_document
                    end
-        transform_key_casing!(document, options[:serialization_context])
+        transform_key_casing!(document, instance_options[:serialization_context])
```

Comment:
@groyoh So, there's no reason the adapter can't use `instance_options` for serialization context (or, really, the options to `initialize` would be the proper place to handle this), but putting the serialization_context in the adapter options isn't incompatible with https://github.com/rails-api/active_model_serializers/pull/1572/files#r56778382